src/ol/webgl/TileTexture.js

```diff
@@ -104,17 +110,19 @@ function uploadDataTexture(
 
   const oldUnpackAlignment = gl.getParameter(gl.UNPACK_ALIGNMENT);
   gl.pixelStorei(gl.UNPACK_ALIGNMENT, unpackAlignment);
+
   gl.texImage2D(
     gl.TEXTURE_2D,
     0,
-    format,
+    internalFormat ? internalFormat : format,
```

Comment:
Float32 should be supported with WebGL as well as WebGL2. Many PCs now running Windows 10 (upgraded from Windows 7) have no WebGL2 support but do support GL_OES_texture_float and GL_OES_texture_float_linear.